wasm/src/graphics/context.rs

```diff
@@ -1,28 +1,32 @@
 use super::render::Render;
+use super::webgl::WebGl2;
 use crate::error::ClientError;
 
 pub struct Context {
-    //render: Render<WebGl>,
+    render: Render<WebGl2>,
 }
 
 impl Context {
     pub fn new() -> Result<Self, ClientError> {
-        //Render::new().map(|render| Self { render })
-        Ok(Self {})
+        Render::new().map(|render| Self { render })
     }
 
     pub fn render(&mut self) -> Result<(), ClientError> {
-        //self.render.render()
-        Ok(())
+        self.render.render()
     }
 }
 
 static mut CONTEXT: Option<Context> = None;
 
-pub fn set_context(context: Context) {
-    unsafe { CONTEXT = Some(context) }
+/// # Safety
+/// This function is not thread safe.
+pub unsafe fn set_context(context: Context) -> &'static mut Context {
```

Comment:
why does this function return anything?

Reply:
See `entries::draw_frame`. Maybe it should be renamed to something like `replace_context`